torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -25,17 +32,45 @@ class ShardingFilterIterDataPipe(IterDataPipe):
     Args:
         source_datapipe: Iterable DataPipe that will be sharded
     """
-    def __init__(self, source_datapipe: IterDataPipe):
+
+    def __init__(self, source_datapipe: IterDataPipe, sharding_group_filter=None):
         self.source_datapipe = source_datapipe
+        self.sharding_group_filter = sharding_group_filter
```

Comment:
Do we need an extra API to set `sharding_group_filter`?

Based on the implementation, it seems `sharding_group_filter` is an integer, could we change it `set` or `list` to support multiple filters?

Reply:
I have no use-cases for it, but it will be trivial to change later if we require to.